### PR TITLE
e2e: add a test case when the GracePeriodSeconds is reach out

### PR DESCRIPTION
**Pull request by zhzhuang-zju**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
refer to #4544, add a test case when the GracePeriodSeconds is reach out in failover_test.go
**Which issue(s) this PR fixes**:
Part of #4544

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhzhuang-zju** [comment]
kindly ping @Vacant2333 @XiShanYongYe-Chang

File: test/e2e/failover_test.go [hunk 1]
```diff
@@ -393,6 +398,87 @@ var _ = framework.SerialDescribe("failover testing", func() {
...
+					})
+			})
+
+			ginkgo.By("check whether application failover with purgeMode graciously when the GracePeriodSeconds is reach out", func() {
+				framework.WaitDeploymentDisappearOnClusters(disabledClusters, deploymentNamespace, deploymentName)
+				evictionTime := time.Now()
+				gomega.Expect(evictionTime.Sub(beginTime) > time.Duration(gracePeriodSeconds+tolerationSeconds)*time.Second).Should(gomega.BeTrue())
+			})
+
+			ginkgo.By("check whether the failed deployment is rescheduled to other available cluster", func() {
```
Comment (XiShanYongYe-Chang) [inline test/e2e/failover_test.go:464]: This block can be removed.

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/lgtm
/approve

**Pull request merged**